lib/matplotlib/axes/_base.py

```diff
@@ -1660,11 +1660,15 @@ def axis(self, *args, emit=True, **kwargs):
                         "3.2", message="Passing more than one positional "
                         "argument to axis() is deprecated and will raise a "
                         "TypeError %(removal)s.")
-                v = args[0]
+                limits = args[0]
                 try:
-                    xmin, xmax, ymin, ymax = v
+                    xmin, xmax, ymin, ymax = limits
                 except ValueError:
                     raise ValueError('args must contain [xmin xmax ymin ymax]')
+                except TypeError:
+                    raise TypeError('the first argument to axis() must be an '
+                                    'interable of the form '
+                                    '[xmin, xmax, ymin, ymax]')
```

Comment:
Not entirely following why one error would be triggered versus the other, but can the same string be used for each?

Reply:
They're different errors; a `TypeError` is thrown if iterable unpacking fails, whereas I'm not actually sure what exact case the previous one is catching.

Reply:
OK, but it seems the error string should say the same thing in both cases?

Reply:
valueerror is raised if then length is not correct.

Reply:
Yeah, I think it's reasonable for the error message to be the same